src/components/information/ScheduledInfoViewer.tsx

```diff
@@ -0,0 +1,133 @@
+import { NonEmptyString, PatternString } from "@pagopa/ts-commons/lib/strings";
+import { Alert } from "design-react-kit";
+import * as iots from "io-ts"; // iots to avoid shadowed-variable on t of i18n
+import moment from "moment";
+import React, { Component } from "react";
+import { WithNamespaces, withNamespaces } from "react-i18next";
+
+/**
+ * String with `YYYY-MM-DDTHH:mm:ss` pattern
+ *
+ * _Example: 2023-07-26T10:05:59_
+ */
+export const ScheduledDateTime = PatternString(
+  "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}$"
+);
+export type ScheduledDateTime = iots.TypeOf<typeof ScheduledDateTime>;
+
+const defaultSeverity = "warning";
+export const SeverityType = iots.union([
+  iots.literal("danger"),
+  iots.literal("info"),
+  iots.literal(defaultSeverity)
+]);
+export type SeverityType = iots.TypeOf<typeof SeverityType>;
+
+export const ScheduledInfo = iots.intersection([
+  iots.type({
+    /**
+     * scheduling start date and time
+     */
+    startDateTime: ScheduledDateTime,
+    /**
+     * scheduling end date and time
+     */
+    endDateTime: ScheduledDateTime,
+    /**
+     * Custom message shown.
+     *
+     * You can set a message or the key of an _i18n_ translation from `scheduledInfo` namespace
+     */
+    message: NonEmptyString
+  }),
+  iots.partial({
+    /**
+     * Define severity color and icon. Default value `warning`.
+     */
+    severity: SeverityType
+  })
+]);
+export type ScheduledInfo = iots.TypeOf<typeof ScheduledInfo>;
+
+export const ScheduledInfoList = iots.array(ScheduledInfo);
+export type ScheduledInfoList = iots.TypeOf<typeof ScheduledInfoList>;
+
+type ScheduledInfoViewerProps = {
+  items: readonly ScheduledInfo[];
+};
+type Props = WithNamespaces & ScheduledInfoViewerProps;
+
+type ScheduledInfoViewerState = {
+  isInRange: boolean;
```

Comment:
Why you need isInRange? Isn't currentMessage !== undefined (or !== empty) enough?
What about a single prop scheduleToDisplay initially undefined that checkScheduledItems sets to something when needed?

Reply:
If you apply the architectural change suggested above you just need the state.

Reply:
The `message`, alone, is not sufficient as a status as I also need the relative `severity` to adequately render the type of `<Alert/>` _(eg: a danger message rather than a warning)._
Moreover, with the boolean `isInRange`, in my opinion, the check for rendering is more readable and immediate.